From 555ba5f3b9cef373fea6d7952618c8999a969db1 Mon Sep 17 00:00:00 2001
From: Zumeng Chen <zumeng.chen@windriver.com>
Date: Sat, 15 Feb 2014 12:35:47 +0800
Subject: [PATCH 001/188] ethernet: cpsw: enable carrier_on after phy_start

Like most ethernet driver, IP level should be well informed when
phy_start is over. If not, it will incur a 100 seconds delay in
the boot process for preempt-rt kernel.

Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/net/ethernet/ti/cpsw.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/drivers/net/ethernet/ti/cpsw.c b/drivers/net/ethernet/ti/cpsw.c
index 4657349..46d0525 100644
--- a/drivers/net/ethernet/ti/cpsw.c
+++ b/drivers/net/ethernet/ti/cpsw.c
@@ -1100,6 +1100,7 @@ static void cpsw_slave_open(struct cpsw_slave *slave, struct cpsw_priv *priv)
 		/* Configure GMII_SEL register */
 		cpsw_phy_sel(&priv->pdev->dev, slave->phy->interface,
 			     slave->slave_num);
+		netif_carrier_on(priv->ndev);
 	}
 }
 
-- 
1.7.5.4

